src/constants/chains.ts

```diff
@@ -0,0 +1,19 @@
+import { arbitrumNovaChainId, gnosisChainId } from '@shapeshiftoss/caip'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import { selectFeatureFlag } from 'state/slices/selectors'
+import { store } from 'state/store'
+// returns known ChainIds as an array, excluding the ones that are currently flagged off
+export const knownChainIds = Object.values(KnownChainIds).filter(chainId => {
+  const isArbitrumEnabled = selectFeatureFlag(store.getState(), 'Arbitrum')
+  const isArbitrumNovaEnabled = selectFeatureFlag(store.getState(), 'ArbitrumNova')
+  const isGnosisEnabled = selectFeatureFlag(store.getState(), 'Gnosis')
+  const isPolygonEnabled = selectFeatureFlag(store.getState(), 'Polygon')
+  const isOptimismEnabled = selectFeatureFlag(store.getState(), 'Optimism')
+  if (chainId === arbitrumNovaChainId && !isArbitrumNovaEnabled) return false
+  if (chainId === KnownChainIds.ArbitrumMainnet && !isArbitrumEnabled) return false
+  if (chainId === gnosisChainId && !isGnosisEnabled) return false
+  if (chainId === KnownChainIds.PolygonMainnet && !isPolygonEnabled) return false
+  if (chainId === KnownChainIds.OptimismMainnet && !isOptimismEnabled) return false
+
+  return true
+})
```

Comment:
looks a bit nicer imo if you are down
```suggestion
import { KnownChainIds } from '@shapeshiftoss/types'
import { selectFeatureFlags } from 'state/slices/selectors'
import { store } from 'state/store'

const enabled = selectFeatureFlags(store.getState())

// returns known ChainIds as an array, excluding the ones that are currently flagged off
export const knownChainIds = Object.values(KnownChainIds).filter(chainId => {
  if (chainId === KnownChainIds.ArbitrumMainnet && !enabled.Arbitrum) return false
  if (chainId === KnownChainIds.ArbitrumNovaMainnet && !enabled.ArbitrumNova) return false
  if (chainId === KnownChainIds.GnosisMainnet && !enabled.Gnosis) return false
  if (chainId === KnownChainIds.OptimismMainnet && !enabled.Optimism) return false
  if (chainId === KnownChainIds.PolygonMainnet && !enabled.Polygon) return false
  return true
})
```